quill-core/src/main/scala/io/getquill/quotation/Quotation.scala

```diff
@@ -9,9 +9,10 @@ import java.util.concurrent.atomic.AtomicInteger
 
 trait Quoted[+T] {
   def ast: Ast
+  def isQuery: Boolean
 }
 
-case class QuotedAst(ast: Ast) extends StaticAnnotation
+case class QuotedAst(ast: Ast, isQuery: Boolean) extends StaticAnnotation
```

Comment:
Would it be possible to avoid the changes here and create specialized `run` methods that return the value instead of list if the input isn't a query?